src/config.rs

```diff
@@ -352,6 +411,8 @@ impl Config {
             sync_once: config.sync_once,
             skip_block_download_wait: config.skip_block_download_wait,
             disable_electrum_rpc: config.disable_electrum_rpc,
+            tx_broadcast_method: config.tx_broadcast_method,
+            tx_broadcast_script: config.tx_broadcast_script,
```

Comment:
It'd be better to have another internal `enum TxBroadcastMethod` that carries `Script(String)` variant and do the conversion where you do the checks.

Reply:
I gave it a try with this commit: d695a222e95ec775bd4976d82f04abab52c4639f

i think adding another type makes the namespacing a bit messier as now we have `pub` types `TxBroadcastMethod`, `TxBroadcastMethodConfigOption`, and `TxBroadcaster`. But it does help us avoid an `unwrap` call in `electrum.rs`, so either works for me.